src/java.base/share/classes/java/util/Objects.java

```diff
@@ -189,6 +189,66 @@ public static String toIdentityString(Object o) {
         return o.getClass().getName() + "@" + Integer.toHexString(System.identityHashCode(o));
     }
 
+    /**
+     * Checks that the specified object reference is an identity object.
+     *
+     * @param obj the object reference to check for identity
+     * @param <T> the type of the reference
+     * @return {@code obj} if {@code obj} is an identity object
+     * @throws NullPointerException if {@code obj} is {@code null}
+     * @throws IdentityException if {@code obj} is not an identity object
+     * @since Valhalla
+     */
+    @ForceInline
+    public static <T> T requireIdentity(T obj) {
+        Objects.requireNonNull(obj);
+        var cl = obj.getClass();
+        if (cl.isValue())
```

Comment:
Just curious, is it that all instances of `Object` (from `new Object()`) are identity, and all other classes that are neither value nor identity are abstract classes?

Reply:
`obj.getClass()` is always concrete and either identity or value. So abstract doesn't really come into play when talking about instances.
This test really should be `!cl.isIdentity()` but that method doesn't exist (yet, TBD soon)

Aabstract classes and interfaces can be either identity or value, or neither.  (But not both).